hydra/core/plugins.py

```diff
@@ -106,33 +106,35 @@ def is_in_toplevel_plugins_module(clazz: str) -> bool:
 
     def instantiate_sweeper(
         self,
-        config: DictConfig,
-        config_loader: ConfigLoader,
+        *,
+        hydra_context: HydraContext,
         task_function: TaskFunction,
+        config: DictConfig,
     ) -> Sweeper:
         Plugins.check_usage(self)
         if config.hydra.sweeper is None:
             raise RuntimeError("Hydra sweeper is not configured")
         sweeper = self._instantiate(config.hydra.sweeper)
         assert isinstance(sweeper, Sweeper)
         sweeper.setup(
-            config=config, config_loader=config_loader, task_function=task_function
+            config=config, hydra_context=hydra_context, task_function=task_function
         )
         return sweeper
 
     def instantiate_launcher(
         self,
-        config: DictConfig,
-        config_loader: ConfigLoader,
+        *,
+        hydra_context: HydraContext,
         task_function: TaskFunction,
+        config: DictConfig,
     ) -> Launcher:
         Plugins.check_usage(self)
         if config.hydra.launcher is None:
             raise RuntimeError("Hydra launcher is not configured")
         launcher = self._instantiate(config.hydra.launcher)
         assert isinstance(launcher, Launcher)
         launcher.setup(
-            config=config, config_loader=config_loader, task_function=task_function
+            config=config, hydra_context=hydra_context, task_function=task_function
         )
         return launcher
```

Comment:
What is the error users will get if their launcher is not upgraded to the new signature?

We could potentially detect it and point them to a migration page.